torch_dftd/functions/dftd3.py

```diff
@@ -247,15 +247,15 @@ def edisp(
 
     if batch_edge is None:
         # (1,)
-        g = e68.sum()[None]
+        g = e68.to(torch.float64).sum()[None]
     else:
         # (n_graphs,)
         if batch.size()[0] == 0:
             n_graphs = 1
         else:
             n_graphs = int(batch[-1]) + 1
-        g = e68.new_zeros((n_graphs,))
-        g.scatter_add_(0, batch_edge, e68)
+        g = e68.new_zeros((n_graphs,), dtype=torch.float64)
+        g.scatter_add_(0, batch_edge, e68.to(torch.float64))
```

Comment:
I found that larger system causes numerical error and the origin is here. Another problem is the nondeterministic behavior of `scatter_add`. To avoid fluctuations during MD, I think it is good to convert values into FP64 explicitly for those specific situations.